x-pack/plugins/security_solution/public/overview/components/detection_response/host_alerts_table/host_alerts_table.tsx

```diff
@@ -0,0 +1,151 @@
+import { HostDetailsLink } from '../../../../common/components/links';
+import { useQueryToggle } from '../../../../common/containers/query_toggle';
+import { useNavigation, NavigateTo, GetAppUrl } from '../../../../common/lib/kibana';
+import * as i18n from '../translations';
+import { LastUpdatedAt, SEVERITY_COLOR } from '../util';
+import { HostAlertsItem, useHostAlertsItems } from './use_host_alerts_items';
+
+type GetTableColumns = (params: {
+  getAppUrl: GetAppUrl;
+  navigateTo: NavigateTo;
+}) => Array<EuiBasicTableColumn<HostAlertsItem>>;
+
+interface HostAlertsTableProps {
+  signalIndexName: string | null;
+}
+
+const DETECTION_RESPONSE_HOST_SEVERITY_QUERY_ID = 'vulnerableHostsBySeverityQuery';
+
+export const HostAlertsTable = React.memo(({ signalIndexName }: HostAlertsTableProps) => {
+  const { getAppUrl, navigateTo } = useNavigation();
+  const { toggleStatus, setToggleStatus } = useQueryToggle(
+    DETECTION_RESPONSE_HOST_SEVERITY_QUERY_ID
+  );
+
+  const { items, isLoading, updatedAt } = useHostAlertsItems({
+    skip: !toggleStatus,
+    queryId: DETECTION_RESPONSE_HOST_SEVERITY_QUERY_ID,
+    signalIndexName,
+  });
+
+  const navigateToHosts = useCallback(() => {
+    navigateTo({ deepLinkId: SecurityPageName.hosts });
+  }, [navigateTo]);
+
+  const columns = useMemo(
+    () => getTableColumns({ getAppUrl, navigateTo }),
+    [getAppUrl, navigateTo]
+  );
+
+  return (
+    <HoverVisibilityContainer show={true} targetClassNames={[INPECT_BUTTON_CLASS]}>
+      <EuiPanel hasBorder data-test-subj="severityHostAlertsPanel">
+        <HeaderSection
+          id={DETECTION_RESPONSE_HOST_SEVERITY_QUERY_ID}
+          title={i18n.HOST_ALERTS_SECTION_TITLE}
+          subtitle={<LastUpdatedAt updatedAt={updatedAt} isUpdating={isLoading} />}
+          titleSize="s"
+          toggleQuery={setToggleStatus}
+          toggleStatus={toggleStatus}
+        />
+        {toggleStatus && (
+          <>
+            <EuiBasicTable
+              items={items}
+              columns={columns}
+              loading={isLoading}
+              data-test-subj="severityHostAlertsTable"
+              noItemsMessage={
+                <EuiEmptyPrompt title={<h3>{i18n.NO_ALERTS_FOUND}</h3>} titleSize="xs" />
+              }
+            />
+            <EuiSpacer size="m" />
+            <EuiButton data-test-subj="severityHostAlertsButton" onClick={navigateToHosts}>
+              {i18n.VIEW_ALL_HOST_ALERTS}
+            </EuiButton>
+          </>
+        )}
+      </EuiPanel>
+    </HoverVisibilityContainer>
+  );
+});
+
+HostAlertsTable.displayName = 'HostAlertsTable';
+
+const getTableColumns: GetTableColumns = ({ getAppUrl, navigateTo }) => [
+  {
+    field: 'hostName',
+    name: i18n.HOST_ALERTS_HOSTNAME_COLUMN,
+    truncateText: true,
+    textOnly: true,
+    render: (hostName: string) => (
+      <HostDetailsLink hostName={hostName} testingId="hostSeverityAlertsTable-hostName" />
+    ),
```

Comment:
You can put the `data-test-subj` in the same TableColumn, so you don't have to add the extra prop to the HostDetailsLink component, I think the test will find the content text even if it is nested inside another element.
```suggestion
    textOnly: true,
    'data-test-subj': 'hostSeverityAlertsTable-hostName',
    render: (hostName: string) => (
      <HostDetailsLink hostName={hostName} />
    ),
```

Reply:
I didnt notice this. Thank you! fixed!